hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiryManager.java

```diff
@@ -94,7 +95,7 @@ public void restartEntitiesTracking() {
 		entityExpiries.reset();
 
 		var expiries = new ArrayList<Map.Entry<Pair<Long, Consumer<EntityId>>, Long>>();
-		schedules.forEach((id, schedule) -> {
+		schedules.get().forEach((id, schedule) -> {
```

Comment:
What happens if the supplier returns null?

Reply:
This can't happen because the `ServiceContext` is [delegating](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java#L2013) to [`ServicesState.scheduleTxns()`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/ServicesState.java#L424); and that child is either [initialized at a genesis startup](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/ServicesState.java#L239), or [ensured non-null](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/ServicesState.java#L192) when restarting from a saved state without a `schedules` FCM.

Reply:
OK. If I understand this right, the reason this fixes the issue is because it defers initialization to some later point in time. We need to file a ticket to address initialization better. The startup order should be perfectly deterministic and clear in the code, otherwise subtle changes to the code can lead to unexpected impacts due to changes in initialization order.

Reply:
It's not so much an initialization issue as it is an identity issue. 

The Platform periodically calls `ServicesState.copy()` to get an immutable copy of the state to sign and save to disk.

We answer that call by [cascading the `copy()`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/ServicesState.java#L346) to all our FCMs. 

But this means that **only** the `ServicesState` instance returned from `copy()` has the mutable (current) versions of those FCMs. All the FCMs in the `ServicesState` instance that received the `copy()` call are now immutable. 

So any component that needs constant access to the latest, mutable version of any FCM must continually call e.g. `ServicesContext.schedules()` to have that request forwarded to the latest, mutable `ServicesState` instance.